src/com/google/javascript/jscomp/Es2020RewriteDynamicImport.java

```diff
@@ -0,0 +1,41 @@
+package com.google.javascript.jscomp;
+
+import com.google.javascript.rhino.Node;
+
+public class Es2020RewriteDynamicImport implements CompilerPass, NodeTraversal.Callback {
+  private final AbstractCompiler compiler;
+  private boolean hasNotified = false;
+
+  static final DiagnosticType DYNAMIC_IMPORT_EXPERIMENTAL =
+      DiagnosticType.warning(
+          "JSC_DYNAMIC_IMPORT_EXPERIMENTAL",
+          "Dynamic import support is experimental. No module rewriting is performed.");
+
+  public Es2020RewriteDynamicImport(AbstractCompiler compiler) {
+    this.compiler = compiler;
+  }
+
+  @Override
+  public void process(Node externs, Node root) {
+    NodeTraversal.traverse(compiler, root, this);
```

Comment:
This should follow the pattern usual to transplation passes.

1. Use `TranspilationPasses.processTranspile()` instead of `NodeTraversal.traverse()`
2. Use `TranspilationPasses.maybeMarkFeaturesAsTranspiledAway()`

See `Es6ForOfConverter#process()` as an example. (Note `Es6RewriteModules` isn't a good pattern to copy, though it may seem tempting since it's related to modules.)

Reply:
Done